### ISSUE TITLE
Install using apt repository doesn't work for fish shell

### ISSUE THREAD

**Issue by AyushSehrawat** [docker/docs#20285]
### Is this a docs issue?

- [X] My issue is about the documentation content or website

### Type of issue

Other

### Description

The `[Install using the apt repository](https://docs.docker.com/engine/install/ubuntu/#install-using-the-repository)` section in documentations shows the commands to add docker repository to apt. Issue is that fish shell doesn't support `$(...)` syntax and gives error

![image](https://github.com/docker/docs/assets/69469790/dbe092ec-8c5f-462e-8dc7-5aa1191ba408)

### Location

https://docs.docker.com/engine/install/ubuntu/

### Suggestion

This can be solved by adding other shell tabs like for fish or bash can be forced to run on that single line using `bash -c` or just a mention to use `bash -c` in case using other shells

**Comment by dvdksn** [comment]
@thaJeztah you're a fish user right?

**Comment by thaJeztah** [comment]
I am; also enough of a Fish user to know that it's not very compatible with POSIX shells, and I usually don't expect such examples to be working without some changes 😅

I didn't try this one on Fish though, as my remote Linux machines tend to be a vanilla install without custom shell.

Not sure we should add a "bash -c" around the whole command to make it work with an incompatible shell (not sure if there's a good solution for this line that would work for both though) 

we should document our conventions for examples to expect a posix(y) shells though.

**Comment by AyushSehrawat** [comment]
> I am; also enough of a Fish user to know that it's not very compatible with POSIX shells, and I usually don't expect such examples to be working without some changes 😅
> 
> I didn't try this one on Fish though, as my remote Linux machines tend to be a vanilla install without custom shell.
> 
> Not sure we should add a "bash -c" around the whole command to make it work with an incompatible shell (not sure if there's a good solution for this line that would work for both though)
> 
> we should document our conventions for examples to expect a posix(y) shells though.

Yeah Bash and fish syntax doesn't always work. How about just a mention to use `bash -c` if not using posix(y) shells? Like a note or something

`bash -c 'command....'`

**Comment by AyushSehrawat** [comment]
@thaJeztah how about a notice like this?

https://cockpit-project.org/running.html#ubuntu

I was browsing through this project and found it. Something similar in documentation will be nice
